File: jorgecunha07-mgt/Mappers/TransportTaskMapper.cs
using System;
using MGT.DTO;
using MGT.Entities;
using MGT.Enums;

namespace MGT.Mappers;

public abstract class TransportTaskMapper
{
    public static TransportTaskDto ToDto(TransportTask transportTask)
    {
        return new TransportTaskDto
        {
            TaskStatus = transportTask.TaskStatus,
            Description = transportTask.Description ?? string.Empty,
            FromLocation = LocationMapper.ToDto(transportTask.FromLocation),
            ToLocation = LocationMapper.ToDto(transportTask.ToLocation),
            ContactStart = transportTask.ContactStart,
            ContactEnd = transportTask.ContactEnd,
            User = transportTask.User,
            RobotId = transportTask.RobotId,
            RobotType = transportTask.RobotType,
            TaskType = transportTask.TaskType,
            ConfirmationCode = transportTask.ConfirmationCode,
            Name = transportTask.Name
        };
    }
    
    public static TransportTask ToEntity(TransportTaskCreateDto taskDto)
    {
        return new TransportTask
        {
            TaskStatus = TaskStatusEnum.Submitted,
            Description = taskDto.Description,
            ContactStart = taskDto.ContactStart,
            ContactEnd = taskDto.ContactEnd,
            FromLocation = LocationMapper.ToEntity(taskDto.FromLocation),
            ToLocation = LocationMapper.ToEntity(taskDto.ToLocation),
            User = taskDto.User,
            RobotId = taskDto.RobotId,
            RobotType = taskDto.RobotType,
            TaskType = TaskTypeEnum.TransportTask,
            Name = taskDto.Name,
            ConfirmationCode = new Random().Next(0000, 9999)
        };
    }
}

File: jorgecunha07-frontend-1dcc492a1c33/jorgecunha07-frontend-1dcc492a1c33/src/app/Mappers/TransportTaskMapper.ts
// Assuming the Location interface and TransportTaskDTO, TransportTask_Model are imported or defined here

import TransportTask_Model from "../models/TransportTask_Model";
import TransportTaskDTO from "../dto/TransportTaskDTO";

export class TransportTaskMapper {
  public static toTransportTaskDTO(model: TransportTask_Model): TransportTaskDTO {
    return {
      Description: model.Description,
      FromLocation: { ...model.FromLocation },
      ToLocation: { ...model.ToLocation },
      ContactStart: model.ContactStart,
      ContactEnd: model.ContactEnd,
      User: model.User,
      RobotId: model.RobotId,
      RobotType: model.RobotType,
      name: model.name
    };
  }

  public static toTransportTaskModel(dto: TransportTaskDTO): TransportTask_Model {
    return {
      Description: dto.Description,
      FromLocation: {
        Building: dto.FromLocation.Building,
        Room: dto.FromLocation.Room,
        X: dto.FromLocation.X,
        Y: dto.FromLocation.Y
      },
      ToLocation: { ...dto.ToLocation },
      ContactStart: dto.ContactStart,
      ContactEnd: dto.ContactEnd,
      User: dto.User,
      RobotId: dto.RobotId,
      RobotType: dto.RobotType,
      name: dto.name
    };
  }
}
